### PR TITLE
#6298 add issue template for new repo

### PR DESCRIPTION
**Pull request by planetf1**
Signed-off-by: Nigel Jones <nigel.l.jones+git@gmail.com>

# Description

Adds new template to support requesting a new repository

See #6298

# How Has This Been Tested?

On a private repo at https://github.com/planetf1/issue-test/issues

# Any additional notes for reviewers?

This update will be merged - it is safe, and only ADDS a new template
It can then be reviewed more easily and changes incorporated
I expect to update it as we work through adding new repos
Overall docs on the repository process/settings are needed -- this is just a first step.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/newrepo.yml [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+          
+            First steps
+            - [ ] Initial code-drop (author)
+          
+            Getting CI/CD started & refining settings
+            - [ ] Add initial build script for PR (including gradle wrapper if required)
+            - [ ] Add initial build script for merge
+          
+            Further Refinement
+            - [ ] Add link in egeria docs to new repo describing purpose etc
```
Comment (davidradl) [inline .github/ISSUE_TEMPLATE/newrepo.yml:151]: maybe include a link into the egeria docs here

File: .github/ISSUE_TEMPLATE/newrepo.yml [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+            - [ ] Add artifact signing if needed
+            - [ ] Add dependabot config        
+            - [ ] Add CodeQL 
+            - [ ] Add to LFXSecurity
+            - [ ] Add to LFAnalytics
+            - [ ] Add check for stale defects
+            - [ ] Add standard issue tags
+          
+            Release
+            - [ ] Add release pipeline
```
Comment (davidradl) [inline .github/ISSUE_TEMPLATE/newrepo.yml:162]: add that the release process should be documented - with the link to the standard place.

File: .github/ISSUE_TEMPLATE/newrepo.yml [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      required: false
+  - type: textarea
+    attributes:
+        label: Work Plan
+        description: | 
+          DO NOT FILL THIS IN NOW
+          
+          This is a checklist to guide the work needed to complete getting the repository and code into an appropriate state for ongoing development. 
+                
+          Where these items may not be undertaken soon after creation, an issue should be opened to track.
```
Comment (davidradl) [inline .github/ISSUE_TEMPLATE/newrepo.yml:130]: Maybe we should document these "DO NOT FILL IN NOW" considerations (or some of these) more advanced / potentially optional checks in egeria-docs and link from the issue template  and (perhaps) not include in the issue template. Being in Egeria docs would mean one place this info is documented and could be searched for.

File: .github/ISSUE_TEMPLATE/newrepo.yml (hunk 1 above)
Comment (planetf1) [reply .github/ISSUE_TEMPLATE/newrepo.yml:151]: Thanks. I'll add a comment to do this in the issue, so we can get the first PR in - let's then work through it with an example,. and I'll collate the updates into the next PR

File: .github/ISSUE_TEMPLATE/newrepo.yml (hunk 3 above)
Comment (planetf1) [reply .github/ISSUE_TEMPLATE/newrepo.yml:130]: I'm unsure.. A simplification perhaps, but I think it's good to be extra clear at the point someone is making the submission. It's easy to miss the detailed docs

The short-term challenge is that those docs don't exist yet (feel free if you'd like to contribute) so a checklist/template was one way to get things moving quickly.

File: .github/ISSUE_TEMPLATE/newrepo.yml (hunk 1 above)
Comment (planetf1) [reply .github/ISSUE_TEMPLATE/newrepo.yml:151]: Adding to https://github.com/odpi/egeria/issues/6298

File: .github/ISSUE_TEMPLATE/newrepo.yml (hunk 3 above)
Comment (planetf1) [reply .github/ISSUE_TEMPLATE/newrepo.yml:130]: Adding to https://github.com/odpi/egeria/issues/6298

File: .github/ISSUE_TEMPLATE/newrepo.yml (hunk 2 above)
Comment (planetf1) [reply .github/ISSUE_TEMPLATE/newrepo.yml:162]: Adding to https://github.com/odpi/egeria/issues/6298

**Pull request merged**